shared/elements/gaia_sim_picker/script.js

```diff
@@ -51,6 +53,21 @@ window.GaiaSimPicker = (function(win) {
     }
   };
 
+  /**
+   * Localize the component manually as l10n attributes are not supported
+   * within the shadow dom. See also: bug 1026236.
+   */
+  proto._localizeShadowDom = function() {
+    var elements = this.shadowRoot.querySelectorAll('[data-l10n-id]');
+    [].forEach.call(elements, el => {
+      var attributes = navigator.mozL10n.getAttributes(el);
+      navigator.mozL10n.formatValue(attributes.id, attributes.args)
+      .then(value => {
+        el.textContent = value;
+      });
+    });
+  };
+
```

Comment:
It would be smarter if you used mozL10n.translateFragment(this.shadowRoot);

translateFragment is what we do on DOM automatically (including DOM Overlays etc.) so the behavior within shadowDOM would match better the one in real DOM.